src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -949,7 +950,10 @@ private static boolean isComparingFieldWithValue(ComparisonPredicate comparisonP
                 || (isFieldPathExpression(rhs) && isValueExpression(lhs));
     }
 
-    private static BsonValue toBsonValue(Object value) {
+    private static BsonValue toLiteralBsonValue(Object value) {
+        if (value instanceof ObjectId) {
+            throw new FeatureNotSupportedException();
+        }
```

Comment:
Thanks for my colleague's refactoring to reuse his `ValueConversions`, but there is one caveat that not all domain values has Bson literal counterpart, noticably `ObjectId`. Also chnaged the method name to make it more explicit and less confusing.

Reply:
How so?  We already have in `ValueConversions` a call to this method:

```
    public static BsonObjectId toBsonValue(ObjectId value) {
        return new BsonObjectId(value);
    }
```

Reply:
Even if `ValueConversions.toBsonValue` did not handle `ObjectId`, which id does, it would have been the responsibility of `ValueConversions.toBsonValue` to throw `SQLFeatureNotSupportedException`, and explicitly checking `instanceof ObjectId` to throw `FeatureNotSupportedException` prematurely would still have been surprising.

Reply:
This method is used to transform query literal domain value to BSON literal values (as shown in `SimpleSelectQueryIntegrationTests#QueryLiteralTests`). For instance, below is an example HQL involving `BigDecimal` (HQL grammar choses `BD` suffix as the indicator):

```
"from Book where price = 123.50BD"
```

However, there is no `ObjectId` query literal recognized by Hibernate. For that reason, I added the exclusion logic to guard against it. It is hard to validate it by testing case for the following possible HQL won't work due to Hibernate's semantic checking:
```
var query = session.createSelectionQuery("from Book where id = :id").setParameter("id", objectId);
```
Currently only `ObjectId` should be put into blacklist, but it troubles me to code for future-proof.

Reply:
> It is hard to validate it by testing case for the following possible HQL won't work due to Hibernate's semantic checking

This sounds like there is no issue at hand that needs to be solved, i.e., no change is needed.